lib/azure_fence.py.py

```diff
@@ -270,10 +270,49 @@ def get_azure_config(options):
 
     return config
 
+# Function to fetch endpoints from metadata endpoint for azure_stack
+def get_cloud_from_arm_metadata_endpoint(arm_endpoint):
+    try:
+        import requests
+        session = requests.Session()
+        metadata_endpoint = arm_endpoint + "/metadata/endpoints?api-version=2015-01-01"
+        response = session.get(metadata_endpoint)
+        if response.status_code == 200:
+            metadata = response.json()
+            return {
+                "resource_manager": arm_endpoint,
+                "credential_scopes": [metadata.get("graphEndpoint") + "/.default"],
+            }
+        else:
+            fail_usage("Failed to get cloud from metadata endpoint: %s - %s" % arm_endpoint, e)
+    except Exception as e:
+        fail_usage("Failed to get cloud from metadata endpoint: %s - %s" % arm_endpoint, e)
+
 def get_azure_cloud_environment(config):
-    cloud_environment = None
-    if config.Cloud:
+    if (config.Cloud is None):
+        config.Cloud = "public"
+
+    try:
+        from azure.mgmt.core import tools
+        import azure.core.settings as settings

```

Comment:
This doesnt work with azure-core 1.24.2, which is the newest version we can use on RHEL8.

Reply:
Thx, I will take a look

Reply:
Yes it's not available in that version :-(.
azure-core 1.24.2 was released on June 30 2022.
The function that I use was introduced in 1.31.0 (September 12 2024)

In the older releases is now azure endpoint handling implemented.

Can you please tell me the version for azure.mgmt.core that is available for Rel8?